jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -595,6 +595,21 @@ public static void readFrom(Path path, ByteBuffer buffer) throws IOException
         }
     }
 
+    public static void readFrom(ReadableByteChannel newReadableByteChannel, long contentLengthValue, ByteBuffer byteBuffer) throws IOException
+    {
+        int pos = BufferUtil.flipToFill(byteBuffer);
+        try (ReadableByteChannel channel = newReadableByteChannel)
```

Comment:
This is going to close the `ReadableByteChannel`, and this should not happen here: the usual convention is to always close an IO resource in the same code context it was opened.

Reply:
Good point, I will move this try with resources to where this method is used.